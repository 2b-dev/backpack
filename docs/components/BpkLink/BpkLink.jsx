import React, { PropTypes } from 'react'
import CssModules from 'react-css-modules'
import { Link } from 'react-router'

import styles from './bpk-link.scss'

const BpkLink = (props) => {
  const { children, to } = props

  return to
    ? (<Link {...props} styleName='bpk-link' activeClassName='bpk-link--active'>{children}</Link>)
    : (<a {...props} styleName='bpk-link'>{children}</a>)
}

BpkLink.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  to: PropTypes.string
}

export default CssModules(BpkLink, styles)
