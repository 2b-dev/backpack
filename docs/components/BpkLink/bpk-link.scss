@import './../../../scss/bonds';
@import './../../../scss/mixins/typography';

.bpk-link {
  @include bpk-link;

  &--active {
    @include bpk-link--active;
  }
}
