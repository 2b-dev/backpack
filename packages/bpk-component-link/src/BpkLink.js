/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2017 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import React from 'react';
import { cssModules } from 'bpk-react-utils';

import STYLES from './bpk-link.scss';

const getClassName = cssModules(STYLES);

const BpkLink = (props) => {
  const {
    children,
    className,
    href,
    onClick,
    blank,
    white,
    ...rest
  } = props;

  const classNames = [getClassName('bpk-link')];
  const target = blank ? '_blank' : null;

  if (white) { classNames.push(getClassName('bpk-link--white')); }
  if (className) { classNames.push(className); }

  return (
    <a className={classNames.join(' ')} href={href} onClick={onClick} target={target} {...rest}>
      {children}
    </a>
  );
};

BpkLink.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  href: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  blank: PropTypes.bool,
  white: PropTypes.bool,
};

BpkLink.defaultProps = {
  className: null,
  onClick: null,
  blank: false,
  white: false,
};

export default BpkLink;
