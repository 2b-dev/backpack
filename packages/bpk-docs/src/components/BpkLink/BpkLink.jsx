import React, { PropTypes } from 'react'
import CssModules from 'react-css-modules'
import { Link } from 'react-router'

import styles from './bpk-link.scss'

const BpkLink = (props) => {
  return props.to
    ? <Link to={props.to} styleName='bpk-link' activeClassName='bpk-link--active' {...props} />
    : <a styleName='bpk-link' {...props} />
}

BpkLink.propTypes = {
  to: PropTypes.string
}

export default CssModules(BpkLink, styles)
