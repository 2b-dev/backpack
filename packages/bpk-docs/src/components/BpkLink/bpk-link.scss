@import '~bpk-mixins/index';

.bpk-link {
  @include bpk-link;

  &--active {
    @include bpk-link--active;
  }
}
