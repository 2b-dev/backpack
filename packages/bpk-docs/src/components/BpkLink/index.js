import component from './BpkLink.jsx'

export default component
