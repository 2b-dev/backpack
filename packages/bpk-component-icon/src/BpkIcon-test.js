jest.unmock('./BpkIcon')

import React from 'react'
import { shallow } from 'enzyme'
import BpkLink from './BpkIcon'

describe('BpkIcon', () => {
})
