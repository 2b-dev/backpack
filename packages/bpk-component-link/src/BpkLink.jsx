import React, { PropTypes } from 'react'

import './bpk-link.scss'

const BpkLink = ({ children, href, onClick }) => (
  <a className='bpk-link' children={children} href={href} onClick={onClick} />
)

BpkLink.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  href: PropTypes.string.isRequired,
  onClick: PropTypes.func
}

BpkLink.defaultProps = {
  onClick: null
}

export default BpkLink
