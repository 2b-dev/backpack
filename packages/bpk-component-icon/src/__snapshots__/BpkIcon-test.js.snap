exports[`BpkIcon should render correctly 1`] = `
<span
  className="bpk-icon-sm"
  dangerouslySetInnerHTML={
    Object {
      "__html": "FILE_MOCK_CONTENTS"
    }
  } />
`;

exports[`BpkIcon should render correctly with a "large" attribute 1`] = `
<span
  className="bpk-icon-lg"
  dangerouslySetInnerHTML={
    Object {
      "__html": "FILE_MOCK_CONTENTS"
    }
  } />
`;
